qcfractal/interface/collections/collection.py

```diff
@@ -180,26 +182,76 @@ def from_json(cls, data: Dict[str, Any], client: "FractalClient" = None) -> "Col
         ret = cls(name, client=client, **data)
         return ret
 
+    @classmethod
+    def from_json(cls, jsondata: Optional[str] = None, filename: Optional[str] = None, client: "FractalClient" = None) -> "Collection":
+        """Creates a new Collection instance from a JSON string.
+
+        Allows roundtrips from `Collection.to_json`.
+
+        One of `jsondata` or `filename` must be provided.
+
+        Parameters
+        ----------
+        jsondata : str, Optional, Default: None
+            The JSON string to create a new Collection instance from.
+        filename : str, Optional, Default: None
+            The filename to read JSON data from.
+        client : FractalClient, optional
+            A FractalClient connected to a server.
+
+        Returns
+        -------
+        Collection
+            A Collection instance.
+
+        """
+        if (jsondata is not None) and (filename is not None):
+            raise ValueError("One of `jsondata` or `filename` must be specified, not both")
+
+        if jsondata is not None:
+            data = json.loads(jsondata)
+        elif filename is not None:
+            with open(filename, 'r') as jsonfile:
```

Comment:
Recommend path lib these days over raw `str`.

Reply:
How do you mean? `pathlib.Path` objects should work with `open` in this way as well. 

I can change the docstring to indicate that a `path-like` is acceptable here; is there a preferred way to do this in the type hints?

Reply:
In light of static/gradual typing I try not to allow ambiguous behavior through which generally forces known types to be those types in stdlib. As in the only allowed option here is a `Path` object, not a string. You could also `Union[str, Path]`, but raw `str` is a bit restrictive when running through MyPy.